samples: serial_lte_modem add MQTT service (#2158)
NordicPlayground/fw-nrfconnect-nrf: samples/nrf9160/serial_lte_modem/README.rst

* AT#XGPS=<op>[,<mask>]

MQTT AT Commands
****************

The following proprietary MQTT AT commands are used in this sample:

* AT#XMQTTCON=<op>[,<url>,<port>[,<sec_tag>]]
* AT#XMQTTPUB=<topic>,<datatype>,<msg>,<qos>,<retain>

> * AT#XMQTTSUB=<topic>,<op>[,<qos>]

Author: Descriptions on #XMQTTEVT and #XMQTTMSG are added to the document.

> * AT#XMQTTSUB=<topic>,<op>[,<qos>]

Author: > How about AT#XMQTTSUB=[,] and AT#XMQTTUNSUB=, as SUBSCRIBE and UNSUBSCRIBE are standard MQTT control packets and commonly used, so it would be good to have two explicit commands

AT#XMQTTSUB is implemented.
Syntax of subscribe/unsubscribe are:
* AT#XMQTTSUB=<topic>,<qos>
* AT#XMQTTUNSUB=<topic>

* AT#XGPS=<op>[,<mask>]

MQTT AT Commands
****************

The following proprietary MQTT AT commands are used in this sample:

* AT#XMQTTCON=<op>[,<url>,<port>[,<sec_tag>]]
* AT#XMQTTPUB=<topic>,<datatype>,<msg>,<qos>,<retain>
* AT#XMQTTSUB=<topic>,<op>[,<qos>]

> The following unsolicited notification reports the MQTT events.

Reviewer: This is hard to understand. Maybe:
```suggestion
The following unsolicited notification indicates that an MQTT event occurred:
```

Author: Thank you. I've modified the description as:

The following unsolicited notification indicates that an MQTT event occurred:

* #XMQTTEVT=<type>,<result>

* #XMQTTEVT=<type>,<result>

> The following unsolicited notification reports the received publish message. The topic and message
> will then be reported.

Reviewer: ```suggestion
The following unsolicited notification indicates that a publish message was received and reports the topic and the message:
```

And does it report the topic and the message, or the topic length and the message length?

Author: The notification will first notify the data type, topic length and message length. Then the topic and message is notified. Carriage Return and Line Feed is inserted before topic and message. The description becomes:

The following unsolicited notification indicates that a publish message was
received and reports the topic and the message:

* #XMQTTMSG=<datatype>,<topic_length>,<message_length><CR><LF><topic><CR><LF><message>

CONFIG_SLM_SUPL_SERVER="supl.google.com"
CONFIG_SLM_SUPL_PORT=7276

> MQTT AT Commands

Author: Done.
